### PR TITLE
Findbugs cleanup

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This PR cleans up the rest of the bugs in the plugin and enforces further static analysis checks

@reviewbybees @jenkinsci/code-reviewers @amuniz

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,8 +16,7 @@
   <properties>
     <jenkins.version>1.609.3</jenkins.version>
     <java.level>6</java.level>
-    <!-- TODO: fix findbugs warnings and remove this -->
-    <findbugs.failOnError>false</findbugs.failOnError>
+    <findbugs.failOnError>true</findbugs.failOnError>
```
Comment (scoheb) [inline pom.xml]: Small thing...is this needed? does it not come from the parent?

Comment (oleg-nenashev) [inline pom.xml]: Yes, makes sense to drop it

**Comment by scoheb** [comment]
+1

File: src/main/java/hudson/plugins/release/dashboard/RecentReleasesPortlet.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ private void rss(StaplerRequest req, StaplerResponse rsp, String suffix, RunList
 		
 		@Extension
 		public static DescriptorImpl newInstance() {
-			if (Hudson.getInstance().getPlugin("dashboard-view") != null) {
+			if (Jenkins.getActiveInstance().getPlugin("dashboard-view") != null) {
```
Comment (jglick) [inline src/main/java/hudson/plugins/release/dashboard/RecentReleasesPortlet.java]: Typically better done with `optional` extensions, but I am not familiar with the specifics here.

Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/release/dashboard/RecentReleasesPortlet.java]: Yes, ideally needs polishing. This plugin just injects extensions

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Pull request merged**